app/http.php

```diff
@@ -0,0 +1,781 @@
+        $log->addTag('verboseType', get_class($error));
+
+        $log->addExtra('file', $error->getFile());
+        $log->addExtra('line', $error->getLine());
+        $log->addExtra('trace', $error->getTraceAsString());
+
+        $log->setAction($action);
+
+        $isProduction = App::getEnv('OPENRUNTIMES_ENV', 'development') === 'production';
+        $log->setEnvironment($isProduction ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+
+        $responseCode = $logger->addLog($log);
+        Console::info('Executor log pushed with status code: ' . $responseCode);
+    }
+
+    Console::error('[Error] Type: ' . get_class($error));
+    Console::error('[Error] Message: ' . $error->getMessage());
+    Console::error('[Error] File: ' . $error->getFile());
+    Console::error('[Error] Line: ' . $error->getLine());
+};
+
+App::post('/v1/runtimes')
+    ->desc("Create a new runtime server")
+    ->param('runtimeId', '', new Text(128), 'Unique runtime ID.')
+    ->param('source', '', new Text(0), 'Path to source files.')
+    ->param('destination', '', new Text(0), 'Destination folder to store build files into.')
+    ->param('vars', '', new Assoc(), 'Environment Variables required for the build')
+    // refactor to `name`
+    ->param('runtime', '', new Text(128), 'Runtime for the cloud function')
+    ->param('baseImage', '', new Text(128), 'Base image name of the runtime')
+    ->inject('orchestrationPool')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, string $source, string $destination, array $vars, string $runtime, string $baseImage, $orchestrationPool, $activeRuntimes, Response $response) {
+
+        // TODO: Check if runtime already exists..
+        $container = 'runtime-' . $runtimeId;
+
+        if ($activeRuntimes->exists($container)) {
+            throw new Exception('Runtime already exists.', 409);
+        }
+
+        $build = [];
+        $buildId = '';
+        $buildStdout = '';
+        $buildStderr = '';
+        $buildStart = \time();
+        $buildEnd = 0;
+
+        try {
+            Console::info('Building runtime with ID : ' . $runtimeId);
+            /** 
+             * Temporary file paths in the executor 
+             */
+            $tmpSource = "/tmp/$runtimeId/code.tar.gz";
+            $tmpBuildDir = "/tmp/$runtimeId/builds";
+            $tmpBuild = "/tmp/$runtimeId/builds/code.tar.gz";
+
+            /**
+             * Copy code files from source to a temporary location on the executor
+             */
+            $device = new Local($destination);
+            $buffer = $device->read($source);
+            if(!$device->write($tmpSource, $buffer)) {
+                throw new Exception('Failed to copy source code to temporary directory', 500);
+            };
+
+            /**
+             * Create a temporary folder to store builds
+             */
+            if (!\file_exists($tmpBuildDir)) {
```

Comment:
I don't think our storage library supports creation of folders. :(

Reply:
I dont think there is concept of folders in S3. Is this still relevant? Should I investigate?

Reply:
The library supports this for local storage automatically, for S3 not relevant. We don't need this check.

https://github.com/utopia-php/storage/blob/main/src/Storage/Device/Local.php#L210

Reply:
@eldadfux  we can't use the `move` method since move both 
1. Creates a folder 
2. Moves the contents of `source` to `target`

In our use case we don't need to copy files to `tmpBuildDir`. We only need to create it and mount it to the container. The **container** then places the built code into this directory.

Reply:
Updated to use storage library methods.